tests/test_discrete.py

```diff
@@ -8,19 +8,32 @@
 
 from powerbox import PowerBox, LogNormalPowerBox, get_power
 
-
 def test_discrete_power_gaussian():
-    pb = PowerBox(N=512,dim=2,boxlength=100., pk = lambda u : 0.1*u**-1.5, ensure_physical=True)
+    # Supply `seed` to ensure that discrete sample corresponds to box.
+    pb = PowerBox(N=512,dim=2,boxlength=100., pk = lambda u : 0.1*u**-1.5,
+        ensure_physical=True, seed=2468)
 
     sample = pb.create_discrete_sample(nbar=1000.)
     power, bins = get_power(sample,pb.boxlength,N=pb.N)
 
-
     res =  np.mean(np.abs(power[50:-50] / (0.1*bins[50:-50]**-1.5) -1 ) )
 
-    print(res)
     assert res < 1e-1
 
+    # This re-grids the discrete sample into a box, basically to verify the
+    # indexing used by meshgrid within `create_discrete_sample`.
+    N = [pb.N] * pb.dim
+    L = [pb.boxlength] * pb.dim
+    edges = [np.linspace(-_L/2., _L/2., _n + 1) for _L, _n in zip(L, N)]
+    delta_samp = np.histogramdd(sample, bins=edges, weights=None)[0].astype("float")
+
+    # Check cross spectrum and assert a strong correlation
+    cross, bins = get_power(delta_samp,pb.boxlength,deltax2=pb.delta_x())
```

Comment:
It's not a big issue, but it's probably a good idea to calculate `delta_x()` once and save it as a variable.

Reply:
Can do @steven-murray, though this makes me think of something else.

Since `create_discrete_sample` calls `delta_x` itself, if one doesn't provide a `seed` initially then the correspondence between tracers and field will be broken for reasons unrelated to this `indexing` keyword argument to `numpy.meshgrid`. Should we be caching realizations to prevent this? I get that this could result in memory bloat if folks are generating many realizations...alternatively, perhaps there should be an optional keyword argument to `create_discrete_samples` to provide the box by hand?

Reply:
Yeah, I had a similar though when I was looking at the code. I'm gonna create an issue to have a look at this in more detail (at some point...)

Reply:
As a placeholder, I've gone the optional keyword argument route, and added a warning if the user doesn't explicitly provide `delta_x` to `create_discrete_samples` AND has not set a random `seed`.